litestar/_openapi/schema_generation/schema.py

```diff
@@ -363,11 +359,40 @@ def for_field_definition(self, field_definition: FieldDefinition) -> Schema | Re
             result = self.for_constrained_field(field_definition)
         elif field_definition.inner_types and not field_definition.is_generic:
             result = self.for_object_type(field_definition)
+        elif field_definition.is_subclass_of(UploadFile):
+            result = self.for_upload_file(field_definition)
         else:
             result = create_schema_for_annotation(field_definition.annotation)
 
         return self.process_schema_result(field_definition, result) if isinstance(result, Schema) else result
 
+    def for_upload_file(self, field_definition: FieldDefinition) -> Schema:
+        """Create schema for UploadFile.
+
+        Args:
+            field_definition: A field definition instance.
+
+        Returns:
+            A Schema instance.
+        """
+
+        if field_definition.is_mapping:
+            return Schema()
+
+        file_schema = Schema(
+            type=OpenAPIType.STRING,
+            content_media_type="application/octet-stream",
+            format=OpenAPIFormat.BINARY,
+        )
+        if field_definition.kwarg_definition is None:
```

Comment:
I'm using `field_definition.kwarg_definition is None` as a proxy to figure out whether this `FieldDefinition` was created directly from the `data` field or whether it was a part of another class. 

So, for the following `field_definition.kwarg_definition` would not be `None`.

```python
@post("/")
async def handle_file_upload(
    data: Annotated[UploadFile, Body(media_type=RequestEncodingType.MULTI_PART)]) -> None:
    ....
```

For the `FieldDefinitions` that reach `for_upload_file` in the below example, the `kwarg_definition` would be `None`.
```python
class Foo(Struct):
    foo: UploadFile
    bar: UploadFile

@post("/f")
async def foo_file(data: Annotated[Foo, Body(media_type=RequestEncodingType.MULTI_PART)]) -> None:
```

The reason this differentiation is needed is because, the populating of `properties` of the schema for `Foo` will be handled by the logic of `for_msgspec_struct` etc. However, if the type is just `UploadFile` like in the first example, then the `properties` needs to be populated within `for_upload_file` itself.

@peterschutt are there any problems with this logic?